generators/utilities/src/main/scala/Simulator.scala

```diff
@@ -1,143 +0,0 @@
-    if (fname.takeRight(2) == ".h") {
-      cfg.simulator match {
-        // verilator needs to explicitly include verilator.h, so use the -FI option
-        case Some(VerilatorSimulator) => s"-FI ${fname}"
-        // vcs pulls headers in with +incdir, doesn't have anything like verilator.h
-        case Some(VCSSimulator) => ""
-        case None => ""
-      }
-    } else { // do nothing otherwise
-      fname
-    }
-  }
-  def writeDotF(lines: Seq[String], cfg: GenerateSimConfig): Unit = {
-    writeTextToFile(lines.mkString("\n"), new File(cfg.targetDir, cfg.dotFName))
-  }
-  // From FIRRTL
-  def safeFile[A](fileName: String)(code: => A) = try { code } catch {
-    case e@ (_: java.io.FileNotFoundException | _: NullPointerException) => throw new Exception(fileName, e)
-    case t: Throwable                                            => throw t
-  }
-  // From FIRRTL
-  def writeResource(name: String, targetDir: String): File = {
-    val in = getClass.getResourceAsStream(name)
-    val p = java.nio.file.Paths.get(name)
-    val fname = p.getFileName().toString();
-
-    val f = new File(targetDir, fname)
-    val out = new java.io.FileOutputStream(f)
-    safeFile(name)(Iterator.continually(in.read).takeWhile(-1 != _).foreach(out.write))
-    out.close()
-    f
-  }
-  // From FIRRTL
-  def writeTextToFile(text: String, file: File) {
-    val out = new java.io.PrintWriter(file)
-    out.write(text)
-    out.close()
-  }
-  def resources(sim: Option[Simulator]): Seq[String] = Seq(
-    "/testchipip/csrc/SimSerial.cc",
-    "/testchipip/csrc/testchip_tsi.cc",
-    "/testchipip/csrc/testchip_tsi.h",
-    "/testchipip/csrc/SimDRAM.cc",
-    "/testchipip/csrc/mm.h",
-    "/testchipip/csrc/mm.cc",
-    "/testchipip/csrc/mm_dramsim2.h",
-    "/testchipip/csrc/mm_dramsim2.cc",
-    "/csrc/SimDTM.cc",
-    "/csrc/SimJTAG.cc",
-    "/csrc/remote_bitbang.h",
-    "/csrc/remote_bitbang.cc",
-    "/vsrc/EICG_wrapper.v",
-    ) ++ (sim match {
-      case None => Seq()
-      case _ => Seq(
-        "/testchipip/csrc/SimSerial.cc",
-        "/testchipip/csrc/SimDRAM.cc",
-        "/testchipip/csrc/mm.h",
-        "/testchipip/csrc/mm.cc",
-        "/testchipip/csrc/mm_dramsim2.h",
-        "/testchipip/csrc/mm_dramsim2.cc",
-        "/csrc/SimDTM.cc",
-        "/csrc/SimJTAG.cc",
-        "/csrc/remote_bitbang.h",
-        "/csrc/remote_bitbang.cc",
-      )
-    }) ++ (sim match { // simulator specific files to include
-      case Some(VerilatorSimulator) => Seq(
-        "/csrc/emulator.cc",
-        "/csrc/verilator.h",
-      )
-      case Some(VCSSimulator) => Seq(
-        "/vsrc/TestDriver.v",
-      )
-      case None => Seq()
-    })
-
-  def writeBootrom(): Unit = {
-    firrtl.FileUtils.makeDirectory("./bootrom/")
-    writeResource("/testchipip/bootrom/bootrom.rv64.img", "./bootrom/")
-    writeResource("/testchipip/bootrom/bootrom.rv32.img", "./bootrom/")
-    writeResource("/bootrom/bootrom.img", "./bootrom/")
```

Comment:
What was this `/bootrom/bootrom.img` in the old code?  I didn't completely understand where it was used in Chipyard before but I knew it wouldn't interfere with what I was doing in FireSim.  I'm only asking because I don't see where this is being maintained in the new makefile version.

Reply:
Good catch. This is Rocket Chips bootrom file: https://github.com/ucb-bar/chipyard/blob/dev/generators/utilities/src/main/resources/bootrom. If you want to use the config fragment in RC (https://github.com/chipsalliance/rocket-chip/blob/b888d0d0eb9975344292d695072e7a9b438cefbb/src/main/scala/subsystem/Configs.scala#L40) you need to have the RC bootrom reside at the top-level in `CHIPYARD/bootrom/bootrom.img`. Ideally RC does what CY does and uses `TargetDirKey`... but for now, this should suffice.

Reply:
I should also note... that 99% of the time people will not use the RC bootrom since the CY one does the same thing for the most part.

Reply:
Ignoring it seems fine to me if they do the same thing.  I didn't realize that's where it was coming from or I would have tried to address that in the PR where I added the `TargetDirKey`.

Reply:
I don't think they quite do the exact same thing anymore. There are some features we support, which the RC bootrom does not, and vice versa.

We should always use the CY bootrom though.

Reply:
In the past we had a `SUB_PROJECT=rocketchip` which setup things to explicitly use RC collateral and not any CY collateral. However, we have drifted from that over time. So I think we can just ignore the RC bootrom.